web/app/project/project.component.spec.ts

```diff
@@ -55,38 +67,248 @@ describe('ProjectComponent', () => {
+         rebuildSummarySubject.next(mockBuildSummary_success);
+
+         expect(component.project.builds.length).toBe(3);
+       }));
   });
 
-  it('should have toolbar with breadcrumbs', () => {
-    fixture.detectChanges();  // onInit()
+  describe('Shallow Tests', () => {
+    describe('Project not loaded', () => {
+      beforeEach(() => {
+        fixture.detectChanges();  // onInit()
+      });
+
+      it('should not show project title', () => {
+        expect(
+            fixture.debugElement.queryAll(By.css('.fci-project-header')).length)
+            .toBe(0);
+      });
+
+      it('table should not show loading spinner', () => {
+        expect(fixture.debugElement
+                   .queryAll(By.css('.fci-build-table .fci-loading-spinner'))
+                   .length)
+            .toBe(1);
+      });
+
+      it('details card should not show loading spinner', () => {
+        expect(
+            fixture.debugElement
+                .queryAll(By.css('.fci-project-details .fci-loading-spinner'))
+                .length)
+            .toBe(1);
+      });
+    });
+
+    describe('Project loaded', () => {
+      beforeEach(fakeAsync(() => {
+        fixture.detectChanges();                // onInit()
+        projectSubject.next(getMockProject());  // Resolve observable
+        flush();
+
+        fixture.detectChanges();
+        flush();
+      }));
+
+      it('should have toolbar with breadcrumbs', () => {
+        // toolbar exists
+        expect(fixture.debugElement.queryAll(By.css('.fci-crumb')).length)
+            .toBe(2);
+
+        expect(component.breadcrumbs[0].label).toBe('Dashboard');
+        expect(component.breadcrumbs[0].url).toBe('/');
+      });
+
+      it('should have project title', () => {
+        expect(fixture.debugElement.query(By.css('.fci-project-header'))
+                   .nativeElement.innerText)
+            .toBe('the most coolest project');
+      });
+
+      describe('table', () => {
+        let rowEls: DebugElement[];
+        beforeEach(() => {
+          rowEls = fixture.debugElement.queryAll(By.css('.mat-row'));
+          expect(rowEls.length).toBe(2);
+        });
+
+        it('should not show loading spinner', () => {
+          expect(fixture.debugElement
+                     .queryAll(By.css('.fci-build-table .fci-loading-spinner'))
+                     .length)
+              .toBe(0);
+        });
+
+        it('should have status icon', () => {
+          expect(rowEls[0]
+                     .query(By.css('.mat-column-number .fci-status-icon'))
+                     .nativeElement.innerText)
+              .toBe('check_circle');
+          expect(rowEls[1]
+                     .query(By.css('.mat-column-number .fci-status-icon'))
+                     .nativeElement.innerText)
+              .toBe('cancel');
+        });
+
+        it('should have build number', () => {
+          expect(rowEls[0]
+                     .query(By.css('.mat-column-number span'))
```

Comment:
is there a better way to address these rows instead of by '.mat-column-number span'? Normally, I'd use some sort of accessibility tag that has a more descriptive phrase (in iOS)

Reply:
Yes we could add an additional class to the span to be more specific. Is that something you're thinking of or was there something else you had in mind?